doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -445,16 +429,23 @@ Requests to the environment
       by_notation ::= @string {? % @scope_key }
```

Comment:
If what I read in `orderedGrammar` is correct, the only use of `by_notation` besides this one is in the `Coercion`, but this command also supports `qualid` in the same position, i.e., it in fact supports `smart_qualid`. Therefore, I suggest to adjust the syntax of `Coercion` and to splice `by_notation` in the definition of `smart_qualid`.

Another remark is that the naming of `smart_qualid` was recently discussed with Hugo and @JasonGross, and they didn't like this name. IIRC they preferred the original (`smart_global`) which I do not like. I proposed `smart_ref` or `smart_reference` as a compromise. I don't remember if there was a conclusion to this discussion.

Reply:
The discussion you are referring to wrapped up with @herbelin 's comment at https://github.com/coq/coq/pull/11840#issuecomment-605447747 which ended with:
> TL;DR
> 
> In the absence of comments, I'm leaning towards continuing exploring the above direction:
> 
> * keeping `reference` to denote something which is parsed as a smart qualid and interpreted as a global reference, adding in passing support for smart references in `Extraction`, `Numeral`, `String`;
> * use `qualid` for current non-term uses of `reference`.

I support this proposal, for the record, which would give us `reference` (what you're currently writing as `smart_qualid`) and `qualid` (as well as `ident` for unqualified identitifers, such as binder names).

Reply:
OK, then for `reference`. Are you fine with this renaming @jfehrle?

Reply:
I will do it.  In the mlgs we have this, which becomes `qualid` in the doc:

```
reference: [
| ident fields
| ident
]
```

which would be a little confusing to code readers (and those reading `common.edit_mlg`) if we have the following in the doc:

```
reference: [
| qualid
| string OPT [ "%" scope_key ]
]
```

FWIW when we later consider adopting some of the doc nonterminal names into the mlgs, we may not want to change some names for compatibility (e.g. for plugins).